flycheck.el

```diff
@@ -5605,6 +5613,14 @@ Hide the error buffer if there is no error under point."
       (save-selected-window
         (quit-window nil window)))))
 
+(defun flycheck-clear-displayed-error-messages ()
+  "Clear error messages displayed by `flycheck-display-error-messages'."
+  (unless (null flycheck--last-displayed-message)
+    (if (and (stringp flycheck--last-displayed-message)
+             (equal (current-message) flycheck--last-displayed-message))
```

Comment:
Why not do a string comparison instead?

Reply:
Because it is the result returned from `display-message-or-buffer`, which can be a string or a window.